include/deal.II/lac/linear_operator.h

```diff
@@ -1111,6 +1111,112 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
   return return_op;
 }
 
+template <typename Range = Vector<double>,
+          typename Domain = Range,
+          typename Matrix>
+LinearOperator<Range, Domain>
+constraints_linear_operator(const ConstraintMatrix &, const Matrix &);
+
+/**
+ * @relates LinearOperator
+ *
+ * constraints_linear_operator takes a constraint matrix @p cm and an exemplar
+ * matrix @p (this matrix could be the system matrix).
+ * The result of this function is the LinearOperator associated to the constraint
+ * matrix.
```

Comment:
I don't think I actually know what this linear operator is. Can you be explicit about what it represents, given a set of constraints $x_i = \sum_j C_{ij} x_j + b_i$?